src/plugins/credential_management/public/components/create_credential_wizard/components/header/add_form.tsx

```diff
@@ -0,0 +1,187 @@
+    errors: [],
+    toasts: [],
+    credentialName: '',
+    description: '',
+    credentialMaterialsType: Credential.CredentialMaterialsType.UsernamePasswordType,
+    userName: '',
+    password: '',
+    dual: true,
+  };
+
+  createCredential = async () => {
+    const { allList }: any = this.props;
+
+    const { credentialName, description, credentialMaterialsType, userName, password } = this.state;
+
+    const validationByField: any = [];
+    if (!credentialName) {
+      validationByField.push('Title should not be empty');
+    }
+    if (allList.find((item: any) => item.title === credentialName)) {
+      validationByField.push('This title is already in use');
+    }
+    if (!description) {
+      validationByField.push('Description should not be empty');
+    }
+
+    if (
+      !userName &&
+      credentialMaterialsType === Credential.CredentialMaterialsType.UsernamePasswordType
+    ) {
+      validationByField.push('User name field should not be empty');
+      this.setState({ nameShowErrors: true });
+    }
+    if (
+      !password &&
+      credentialMaterialsType === Credential.CredentialMaterialsType.UsernamePasswordType
+    ) {
+      validationByField.push('Password field should not be empty');
+      this.setState({ nameShowErrors: true });
+    }
+
+    if (credentialMaterialsType === 'no_auth') {
+      this.setState({ nameShowErrors: false });
+    }
+
+    if (validationByField.length) {
+      this.setState({ showErrors: true, errors: validationByField });
+      return;
+    }
+
+    this.setState({ showErrors: false, errors: [] });
+    this.props.onLoading();
+    const { savedObjects }: any = this.props;
+
+    try {
+      await savedObjects.client.create('credential', {
+        title: this.state.credentialName,
+        description: this.state.description,
+        credentialMaterials: {
+          credentialMaterialsType: this.state.credentialMaterialsType,
+          credentialMaterialsContent: {
+            username: this.state.userName,
+            password: this.state.password,
+          },
+        },
+      });
+      this.props.onHispush();
+    } catch (e) {
+      const createCredentialFailMsg = (
+        <FormattedMessage
+          id="credentialManagement.createCredential.loadCreateCredentialFailMsg"
+          defaultMessage="The credential saved object creation failed with some errors. Please configure data_source.enabled and try it again."
+        />
+      );
+      this.setState((prevState: any) => ({
+        toasts: prevState.toasts.concat([
+          {
+            title: createCredentialFailMsg,
+            id: createCredentialFailMsg.props.id,
+            color: 'warning',
+            iconType: 'alert',
+          },
+        ]),
+      }));
+    }
+    this.props.onCancelLoading();
+  };
+
+  render() {
+    const { showErrors, errors, nameShowErrors } = this.state;
+
+    const options = [
+      {
+        value: Credential.CredentialMaterialsType.UsernamePasswordType,
```

Comment:
Suggestion: as these options are constant and not changing .. can we have these in constants file like Credential.CredentialMaterialsType.UsernamePasswordType ?